test/e2e/policy/federatedresourcequota_test.go

```diff
@@ -0,0 +1,135 @@
+
+		clusterName := "member-e2e-" + rand.String(3)
+		homeDir := os.Getenv("HOME")
+		kubeConfigPath := fmt.Sprintf("%s/.kube/%s.config", homeDir, clusterName)
+		controlPlane := fmt.Sprintf("%s-control-plane", clusterName)
+		clusterContext := fmt.Sprintf("kind-%s", clusterName)
+
+		ginkgo.BeforeEach(func() {
+			framework.CreateFederatedResourceQuota(e2e.KarmadaClient, federatedResourceQuota)
+		})
+
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("Creating cluster: %s", clusterName), func() {
+				err := e2e.CreateCluster(clusterName, kubeConfigPath, controlPlane, clusterContext)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By(fmt.Sprintf("Unjoinning cluster: %s", clusterName), func() {
+				karmadaConfig := karmadactl.NewKarmadaConfig(clientcmd.NewDefaultPathOptions())
+				opts := karmadactl.CommandUnjoinOption{
+					GlobalCommandOptions: options.GlobalCommandOptions{
+						DryRun: false,
+					},
+					ClusterNamespace:  "karmada-cluster",
+					ClusterName:       clusterName,
+					ClusterContext:    clusterContext,
+					ClusterKubeConfig: kubeConfigPath,
+					Wait:              options.DefaultKarmadactlCommandDuration,
+				}
+				err := karmadactl.RunUnjoin(os.Stdout, karmadaConfig, opts)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By(fmt.Sprintf("Deleting clusters: %s", clusterName), func() {
+				err := e2e.DeleteCluster(clusterName, kubeConfigPath)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+				_ = os.Remove(kubeConfigPath)
+			})
+		})
+
+		ginkgo.AfterEach(func() {
+			framework.RemoveFederatedResourceQuota(e2e.KarmadaClient, frqNamespace, frqName)
+		})
+
+		ginkgo.It("federatedResourceQuota should be propagated to new joined clusters", func() {
+			ginkgo.By(fmt.Sprintf("Joinning cluster: %s", clusterName), func() {
+				karmadaConfig := karmadactl.NewKarmadaConfig(clientcmd.NewDefaultPathOptions())
+				opts := karmadactl.CommandJoinOption{
+					GlobalCommandOptions: options.GlobalCommandOptions{
+						DryRun: false,
+					},
+					ClusterNamespace:  "karmada-cluster",
+					ClusterName:       clusterName,
+					ClusterContext:    clusterContext,
+					ClusterKubeConfig: kubeConfigPath,
+				}
+				err := karmadactl.RunJoin(os.Stdout, karmadaConfig, opts)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+
+			ginkgo.By(fmt.Sprintf("waiting federatedResourceQuota(%s/%s) present on cluster: %s", frqNamespace, frqName, clusterName), func() {
+				clusterClient, err := util.NewClusterClientSet(clusterName, e2e.ControlPlaneClient, nil)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+				gomega.Eventually(func(g gomega.Gomega) (bool, error) {
+					_, err := clusterClient.KubeClient.CoreV1().ResourceQuotas(frqNamespace).Get(context.TODO(), frqName, metav1.GetOptions{})
```

Comment:
the data of  resource quota is also checked

Reply:
I plan to add a case to test status collection, it will cover this situation.